### PR TITLE
add background events

### PR DESCRIPTION
**Pull request by bgates**
Hi, I was interested in the suggestion by @dmoskovsov in #103 so I tried to resolve the concerns you had about the API there.  An event object has a Boolean flag to mark it as `background`; the default behavior is to place the title of a background event in the same row as the event date on the month calendar, but it's also possible to set up a separate component and prop getter to structure and style the background event.

### CODE REVIEW

**Comment by jquense** [comment]
hey there, thanks a lot for the PR! I'm excited to take a look at this but on vacation this week so there may be a delay.  thanks!

File: src/BackgroundCells.jsx [hunk 1]
```diff
@@ -33,20 +39,43 @@ class DisplayCells extends React.Component {
...
+      let backgroundEvent = backgroundEvents.find(event => {
+        return dates.sameDay(event.start, week[i])
+      })
+
+      let Component = backgroundEventComponent;
+      let title = (backgroundEvent || {}).title;
+
+      var style, xClassName;
+      if (backgroundEvent && backgroundEventPropGetter)
+        ({ style, className: xClassName } = backgroundEventPropGetter(backgroundEvent));
```
Comment (jquense) [inline src/BackgroundCells.jsx]: `let { style, className: xClassName } = backgroundEventPropGetter(backgroundEvent) || {};`

File: src/BackgroundCells.jsx [hunk 2]
```diff
@@ -33,20 +39,43 @@ class DisplayCells extends React.Component {
...
+          style={{...segStyle(1, slots), ...style, ...props.style}}
+          className={cn('rbc-day-bg', xClassName, {
             'rbc-selected-cell': selecting && i >= startIdx && i <= endIdx
           })}
-        />
+        >
+        { Component && backgroundEvent
+          ? <Component
+              event={backgroundEvent}
+              eventPropGetter={backgroundEventPropGetter}
```
Comment (jquense) [inline src/BackgroundCells.jsx]: this has already been used above, I wouldn't pass it down further if the prop has been consumed

**Comment by jquense** [comment]
hi there 👋  thanks again for the PR. I think there are probably two different features here, which are each a bit confused. 

1. styling background cells 
2. adding "background events" or labeled regions.

I think this PR would work great for 1. (the original issue) but has some issues with being a solution for 2. The big one is that there is no support for overlapping background events, or handling of the styling for the time views (day, week, etc).

I think it probably makes sense to roll back the back event stuff for now and just go with the backgroundPropGetter. Implementing 2. should probably involve some of the talk in https://github.com/intljusticemission/react-big-calendar/issues/115

does that make sense?

File: src/BackgroundCells.jsx (hunk 1 above)
Comment (bgates) [inline src/BackgroundCells.jsx]: That assumes there's a `backgroundEventPropGetter`, doesn't it?

**Comment by bgates** [comment]
I guess I'd call them 'interrelated' features rather than 'confused' 😄 . The problem I see with separating them is that I don't know why the background would be styled differently in the absence of a background event.

**Pull request closed without merging**